File: file_processing/pl/desc_numeric_file_sort.pl
#!/usr/bin/perl

###################################################
## slurp lines from file of numbers 
## and sort them numerically using the default sort.
## 
## To run:
## >> perl desc_numeric_file_sort.pl numbers.txt
## @author: vladimir kulyukin
###################################################
use strict;
use warnings;

# 1. get $ARGV[0]
my $infile_path = shift;
# 2. open $file_path for reading or die
open(IN, '<', $infile_path) or die $!;
# 3. slurp the lines, sort, print
my @lines = <IN>;
print sort { $b <=> $a } @lines;

File: file_processing/py/desc_numeric_file_sort.py
#!/usr/bin/python

#######################################
# Read in a file of numbers and
# sort them from largest to smallest.
# @author: vladimir kulyukin
#######################################

import sys

# 1. get 1st command line argument
file_path = sys.argv[1]
# 2. open file_path for reading
infile = open(file_path, 'r')
# 3. read in the lines, sort, print
for n in sorted([int(line) for line in infile.readlines()],
                reverse=True):
    print n
